feat: UIG-2178 - header footer ready event

ksvlaanderen opened the pull request:
ready event toegevoegd voor header en footer
refactor header en footer
verbeteringen Storybook en Cypress
ecmascript versie verhoogd

Goldflow commented on libs/sections/src/lib/header/vl-header.section.ts:
```diff
@@ -30,63 +27,110 @@ export class VlHeader extends LitElement {
             ? `https://tni.widgets.burgerprofiel.dev-vlaanderen.be/api/v1/widget/${this.identifier}`
             : `https://prod.widgets.burgerprofiel.vlaanderen.be/api/v1/widget/${this.identifier}`;
 
         (window as any).vl.widget.client
```
ik denk niet dat in de scope van dit ticket hoort; maar misschien eens goed op een bepaald moment het `vl` object op de `window` te typen zodat we gaandeweg van onze `any`'s afkunnen

Goldflow replied:
Kristof heeft dit opgenomen in "waar we naar toe willen document"

Goldflow commented on libs/sections/src/lib/header/vl-header.section.ts:
```diff
@@ -1,21 +1,18 @@
     awaitScript(
-        'vl-header-client',
+        'vl-header-polyfill',
         'https://prod.widgets.burgerprofiel.vlaanderen.be/api/v1/node_modules/@govflanders/vl-widget-polyfill/dist/index.js'
```
mss een idee om deze url als constanten / globale variabelen in te stellen en die vanuit een aparte file te laten importeren?

zelfde commentaar voor equivalente code in `vl-footer.section.ts`

Goldflow replied:
we gaan dit niet doen, gezien die url's vrijwel nooit gaan veranderen - besproken

Goldflow commented on libs/sections/src/lib/header/vl-header.section.ts:
```diff
@@ -30,63 +27,110 @@ export class VlHeader extends LitElement {
+    private observeWidgetIsAdded() {
+        const isHeader = (node: Node) => {
+            return (
+                (node as HTMLElement).tagName === 'HEADER' || (node.childNodes && [...node.childNodes].some(isHeader))
```
als ik dit goed begrijp kan de afnemer hier ook met `<header>` tag werken?

Goldflow replied:
besproken; dit is gedaan op de manier die onze voorgangers ook gedaan hebben
widget van Digitaal Vlaanderen heeft `<header>` tag

Goldflow reviewed (approved):
enkele opmerkingen maar ziet er verder goed uit

wou mss even toelichting bij de recursieve functie `isHeader` maar om een tree like structuur te doorlopen it makes sense? misschien vraag ik je later nog even wat toenadering

Goldflow commented on libs/sections/src/lib/header/stories/vl-header.stories-doc.mdx:
```diff
@@ -0,0 +1,38 @@
+import { ArgsTable, DocsStory, PRIMARY_STORY } from '@storybook/addon-docs';
```
`@storybook/addon-docs` is zover ik kan zien missing bij `package.json`

ksvlaanderen replied:
Dit komt standaard met Storybook mee.

Goldflow replied:
ok!

Goldflow commented on libs/sections/src/lib/footer/stories/vl-footer.stories-doc.mdx:
```diff
@@ -0,0 +1,38 @@
+import { ArgsTable, DocsStory, PRIMARY_STORY } from '@storybook/addon-docs';
```
zelfde hier, kan je even `@storybook/addon-docs` toevoegen bij `package.json`?

ksvlaanderen replied:
Dit komt standaard met Storybook mee.

Goldflow replied:
ok goed!

Goldflow commented on libs/sections/src/lib/footer/stories/vl-footer.stories-doc.mdx:
```diff
@@ -0,0 +1,38 @@
+
+### Digitaal Vlaanderen
+
+**Documentatie Digitaal Vlaanderen:** https://vlaamseoverheid.atlassian.net/wiki/spaces/IKPubliek/pages/2038137621/Technische+documentatie+global+header+-+global+footer#Integratie-global-header-met-eenvoudig-toegangsbeheer-(ACM)
```
mss beter deze link te gebruiken (niet zeker)

https://www.vlaanderen.be/digitaal-vlaanderen/onze-oplossingen/mijn-burgerprofiel/koppelen-met-mijn-burgerprofiel-als-dienstenleverancier/technische-toolkit-voor-aansluitingen-door-dienstenleveranciers

deze link geeft ook wat meer context (oa hoe de aanvraag te starten voor je specifieke app) + linkt ook naar deze de link die je al had & een andere technische documentatie

ditzelfde comment geld ook voor header natuurlijk

ksvlaanderen replied:
Aangepast.

Goldflow commented on libs/sections/src/lib/header/stories/vl-header.stories-doc.mdx:
```diff
@@ -0,0 +1,38 @@
+
+### Digitaal Vlaanderen
+
+**Documentatie Digitaal Vlaanderen:** https://vlaamseoverheid.atlassian.net/wiki/spaces/IKPubliek/pages/2038137621/Technische+documentatie+global+header+-+global+footer#Integratie-global-header-met-eenvoudig-toegangsbeheer-(ACM)
```
zie gelijkaardige comment bij footer

ksvlaanderen replied:
Aangepast.

Goldflow reviewed (changes requested):
- commentaar op de links ben ik niet zeker; maar lijkt me persoonlijk wel goed
- npm packages, op 1 of andere manier vind ik die niet terug in de `package.json`

Commit:
feat: UIG-2178 / UIG-2260 - header en footer

ready event header en footer
refactor header en footer
verbeteringen Storybook en Cypress
ecmascript versie verhoogd

kspeltix commented:
:tada: This PR is included in version 1.1.0 :tada:

The release is available on [GitHub release](https://github.com/milieuinfo/uigov-web-components/releases/tag/v1.1.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
